src/binding/c/misc_api.txt

```diff
@@ -0,0 +1,200 @@
+                               "MPICH CC:\t%s\n"
+                               "MPICH CXX:\t%s\n"
+                               "MPICH F77:\t%s\n"
+                               "MPICH FC:\t%s\n",
+                               MPII_Version_string, MPII_Version_date, MPII_Version_ABI,
+                               MPII_Version_device, MPII_Version_configure, MPII_Version_CC,
+                               MPII_Version_CXX, MPII_Version_F77, MPII_Version_FC);
+    if (strlen(MPII_Version_custom) > 0)
+        MPL_snprintf(version + printed_len, MPI_MAX_LIBRARY_VERSION_STRING - printed_len,
+                     "MPICH Custom Information:\t%s\n", MPII_Version_custom);
+
+    *resultlen = (int) strlen(version);
+}
+
+MPI_Pcontrol:
+    .desc: Controls profiling
+    .skip: ThreadSafe, validate-PROFILE_LEVEL, validate-VARARGS
+    .impl: direct
+{
+    int mpi_errno = MPI_SUCCESS;
+    va_list list;
+    MPIR_FUNC_TERSE_STATE_DECL(MPID_STATE_MPI_PCONTROL);
+
+    MPIR_ERRTEST_INITIALIZED_ORDIE();
+
+    MPIR_FUNC_TERSE_ENTER(MPID_STATE_MPI_PCONTROL);
+
+    /* ... body of routine ...  */
+
+    /* This is a dummy routine that does nothing.  It is intended for
+     * use by the user (or a tool) with the profiling interface */
+    /* We include a reference to va_start and va_end to (a) quiet some
+     * compilers that warn when they are not present and (b) show how to
+     * access any optional arguments */
+    va_start(list, level);
+    va_end(list);
+
+    /* ... end of body of routine ... */
+    MPIR_FUNC_TERSE_EXIT(MPID_STATE_MPI_PCONTROL);
+    return mpi_errno;
+    /* There should never be any fn_fail case; this suppresses warnings from
+     * compilers that object to unused labels */
+}
+
+MPI_Get_version:
+    .desc: Return the version number of MPI
+    .skip: ThreadSafe, initcheck
+/*
+    .N SignalSafe
+*/
+{
+    *version = MPI_VERSION;
+    *subversion = MPI_SUBVERSION;
+}
+
+MPIX_GPU_query_support:
+    gpu_type: GPU_TYPE, [MPIX_GPU_SUPPORT_CUDA or MPIX_GPU_SUPPORT_ZE]
+    is_supported: LOGICAL, direction=out, [true if gpu of given type is supported, false otherwise.]
+    .desc: Returns whether the specific type of GPU is supported
+    .skip: validate-GPU_TYPE
+    .docnotes: SignalSafe
+{
+    *is_supported = 0;
+    if (MPIR_CVAR_ENABLE_GPU) {
+        MPL_gpu_type_t type;
+        MPL_gpu_query_support(&type);
+
+        switch (gpu_type) {
+            case MPIX_GPU_SUPPORT_CUDA:
+                if (type == MPL_GPU_TYPE_CUDA)
+                    *is_supported = 1;
+                break;
+
+            case MPIX_GPU_SUPPORT_ZE:
+                if (type == MPL_GPU_TYPE_ZE)
+                    *is_supported = 1;
+                break;
+
+            default:
+                MPIR_ERR_SETANDJUMP(mpi_errno, MPI_ERR_ARG, "**badgputype");
+        }
+    }
+}
+
+MPIX_Query_cuda_support:
+    .desc: Returns whether CUDA is supported
+    .impl: direct
+    .docnotes: SignalSafe
```

Comment:
I believe it returns the support directly. Can we stop emitting `.N Errors` and says "Return value: support of CUDA" etc?

(I understand that this routine is not part of the official MPI).

Reply:
Skipped `Errors`. 

I believe the short description and function prototype is sufficient to indicate what is the return value.